File: src/main/java/com/sunkenquest/ExpenseTracker/sevices/income/IncomeService.java
package com.sunkenquest.ExpenseTracker.sevices.income;

import org.springframework.data.domain.Page;

import com.sunkenquest.ExpenseTracker.dto.IncomeDTO;
import com.sunkenquest.ExpenseTracker.entity.Income;

public interface IncomeService {
    Income postIncome(IncomeDTO incomeDTO);

    Page<IncomeDTO> getAllIncomes(Integer page);

    Income updateIncome(Long id, IncomeDTO incomeDTO);

    IncomeDTO getIncomeById(Long id);

    void deleteIncomeById(Long id);
}

File: frontend/src/app/services/income/income.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

const BASIC_URL = "http://localhost:8080/";

@Injectable({
  providedIn: 'root'
})
export class IncomeService {

  constructor(
    private http: HttpClient
  ) { }

  postIncome(incomeDTO: any): Observable<any> {
    return this.http.post(BASIC_URL + "api/income", incomeDTO)
  }

  getAllIncomes(page: number): Observable<any> {
    return this.http.get(BASIC_URL + `api/income/all?page=${page}`)
  }

  getIncomeById(id: number): Observable<any> {
    return this.http.get(BASIC_URL + `api/income/${id}`)
  }

  updateIncome(id: number, incomeDTO: any): Observable<any> {
    return this.http.put(BASIC_URL + `api/income/${id}`, incomeDTO
    )
  }

  deleteIncome(id: number): Observable<any> {
    return this.http.delete(BASIC_URL + `api/income/${id}`)
  }
}
